server/server.go

```diff
@@ -1068,6 +1071,77 @@ func writeHealthResponse(w http.ResponseWriter, err error, payload interface{})
 	writer.JSON(w, status, payload, false)
 }
 
+func (s *Server) unversionedGetHealthWithPolicy(w http.ResponseWriter, r *http.Request) {
+	pluginStatus := s.manager.PluginStatus()
+	pluginState := map[string]string{}
+
+	// optimistically assume all plugins are ok
+	allPluginsOk := true
+	// iterate over plugin status to extract state
+	for name, status := range pluginStatus {
+		if status != nil {
+			pluginState[name] = string(status.State)
+			// if all plugins have not been in OK state yet, then check to see if plugin state is OKx
+			if !s.allPluginsOkOnce && status.State != plugins.StateOK {
+				allPluginsOk = false
+			}
+		}
+	}
+	// once all plugins are OK, set the allPluginsOkOnce flag to true, indicating that all
+	// plugins have achieved a "ready" state at least once on the server.
+	if allPluginsOk {
+		s.allPluginsOkOnce = true
+	}
+
+	input := map[string]interface{}{
+		"plugin_state":  pluginState,
+		"plugins_ready": s.allPluginsOkOnce,
+	}
+
+	vars := mux.Vars(r)
+	urlPath := vars["path"]
+	healthDataPath := fmt.Sprintf("/system/health/%s", urlPath)
+	healthDataPath = stringPathToDataRef(healthDataPath).String()
+
+	rego := rego.New(
+		rego.Query(healthDataPath),
+		rego.Compiler(s.getCompiler()),
+		rego.Store(s.store),
+		rego.Input(input),
+		rego.Runtime(s.runtime),
+	)
+
+	rs, err := rego.Eval(r.Context())
+
+	if err != nil {
+		writeHealthPolicyResponse(w, err)
+		return
+	}
+
+	if len(rs) == 0 {
+		writeHealthPolicyResponse(w, fmt.Errorf("Health policy failed at %s", healthDataPath))
```

Comment:
nit: by convention error messages start with lower-case not upper case. What about "health policy was undefined"?